api_tests/src/user.spec.ts

```diff
@@ -95,3 +97,20 @@ test("Delete user", async () => {
   expect((await resolvePost(alpha, remotePost)).post).toBeUndefined();
   expect((await resolveComment(alpha, remoteComment)).comment).toBeUndefined();
 });
+
+test("Requests with invalid auth should be treated as unauthenticated", async () => {
+  let invalid_auth: API = {
+    client: new LemmyHttp("http://127.0.0.1:8541"),
+    auth: "invalid",
+  };
+  let site = await getSite(invalid_auth);
+  expect(site.my_user).toBeUndefined();
+  expect(site.site_view).toBeDefined();
+
+
+  let form: GetPosts = {
+    auth: invalid_auth.auth,
+  };
+  let posts = invalid_auth.client.getPosts(form);
+  expect((await posts).posts).toBeDefined();
+});
```

Comment:
This is a major improvement, read-only api calls now treat the user as unauthenticated in case of invalid auth or site ban. This partially solves https://github.com/LemmyNet/lemmy-ui/issues/989.

Reply:
Nice.